src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -158,7 +158,14 @@ private ImmutableArray<BoundExpression> LeftHandSideSideEffects(ImmutableArray<B
 
             foreach (var variable in variables)
             {
-                lhsReceivers.Add(TransformCompoundAssignmentLHS(variable, stores, temps, isDynamicAssignment: variable.Type.IsDynamic()));
+                if (variable.Kind == BoundKind.DiscardedExpression)
+                {
+                    lhsReceivers.Add(variable);
```

Comment:
The side-effects are `stores`. 
`lhsReceivers` contains the receivers for the final assignment step. Discards receivers are handled in `ApplyAssignments` above (ie. they are skipped).

---
In reply to: [88564795](https://github.com/dotnet/roslyn/pull/15192#discussion_r88564795) [](ancestors = 88564795)